secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/account/v4_0_0/statements/StatementsApiController.java

```diff
@@ -159,8 +170,38 @@ public ResponseEntity<Object> getAccountsAccountIdStatementsStatementIdFile(Stri
     }
 
     @Override
-    public ResponseEntity<OBReadTransaction6> getAccountsAccountIdStatementsStatementIdTransactions(String statementId, String accountId, String authorization, String xFapiAuthDate, String xFapiCustomerIpAddress, String xFapiInteractionId, String xCustomerUserAgent, String apiClientId, String consentId, int page) {
-        return null;
+    public ResponseEntity<OBReadTransaction6> getAccountsAccountIdStatementsStatementIdTransactions(String statementId, String accountId, String authorization, String xFapiAuthDate, LocalDateTime fromBookingDateTime, LocalDateTime toBookingDateTime, String xFapiCustomerIpAddress, String xFapiInteractionId, String xCustomerUserAgent, String apiClientId, String consentId, int page)
+            throws OBErrorException {
+        logger.info("getAccountStatementTransactions from account id {}, statement id {}, consentId: {}, apiClientId: {} fromBookingDate {} toBookingDate {} pageNumber {} ",
+                    accountId, statementId, consentId, apiClientId, fromBookingDateTime, toBookingDateTime, page);
+
+        if (toBookingDateTime == null) {
+            toBookingDateTime = LocalDateTime.now();
+        }
+        if (fromBookingDateTime == null) {
+            fromBookingDateTime = toBookingDateTime.minus(Period.ofYears(100));
```

Comment:
Does this make sense? 100 years?

Reply:
Same as in v3.1.10, I don't know why this was the way it was implemented but I asumed it was correct there :)

Reply:
Most likely because we control the test users and their number of transactions, and also we have pagination which is limiting the number of entries returned in one go. So this 100 years period is to make sure all entries are returned but the pagination is taking care of the number of entries.

Reply:
Sure, ok. 

I guess it's not a problem for now, being 100 years. But just looking up the last x years means over time we'd start losing data. A fixed date is better.

Ok for now though.